### ISSUE TITLE
v7.29 error in new profile yaml loading

### ISSUE THREAD

**Issue by ASLeonard** [snakemake/snakemake#2326]
After upgrading to `v7.29.0` and running a command like `snakemake --profile "Euler"`, I now get the following error

```
snakemake: error: Couldn't parse config file: 'tuple' object has no attribute 'safe_load'
```

downgrading to `v7.28.3` works again, and it looks like the likely culprit is this change introduced in `v7.29.0`

https://github.com/snakemake/snakemake/blob/f1b8342757fc5554eb7a9ed5d393c376e9089c3c/snakemake/profiles.py#L11C4-L11C48

Not sure of the deeper logic, but changing
```
yaml = self._load_yaml()
```
to just 
```
import yaml
```

made it through. The original yaml variable has the value of 
```
(<module 'yaml' from '.../miniconda3/envs/snake/lib/python3.11/site-packages/yaml/__init__.py'>, <class 'yaml.cyaml.CSafeLoader'>, <class 'yaml.cyaml.CDumper'>)
```
so alternatively can do 
```
yaml = self._load_yaml()[0]
```

**Comment by edwinyyyu** [comment]
Looks like this caused it:
https://github.com/bw2/ConfigArgParse/commit/651d7d0260d65576857c0cf6c2bc424d3580a5a9

**Comment by corneliusroemer** [comment]
This looks could be a serious bug. I got the same error and it took me quite a bit of hunting to find this issue which doesn't seem to be indexed by Google yet.
I've added a quick Stackoverflow Q&A for others who will inevitably run into this soon: https://stackoverflow.com/questions/76573531/when-using-profile-get-error-snakemake-error-couldnt-parse-config-file-tu/76573532#76573532
(ping @johanneskoester 🙏 )

**Comment by corneliusroemer** [comment]
I confirm it's fixed for me with 7.30.0
